cg/meta/demultiplex/demux_post_processing.py

```diff
@@ -151,39 +149,34 @@ def is_flow_cell_directory_valid(self, flow_cell_directory: Path) -> bool:
             LOG.warning(f"Flow cell directory does not exist: {flow_cell_directory}")
             return False
 
-        if not Path(flow_cell_directory, DemultiplexingDirsAndFiles.DEMUX_COMPLETE).exists():
+        if not self.is_demultiplexing_complete(flow_cell_directory=flow_cell_directory):
             LOG.warning(f"Demultiplexing is not complete for flow cell {flow_cell_directory.name}")
             return False
 
         return True
 
-    def update_sample_read_counts(self, sample_internal_ids: List[str]) -> None:
-        """Update samples in status db with read counts from the SampleLaneSequencingMetrics table."""
+    def is_demultiplexing_complete(self, flow_cell_directory: Path) -> bool:
+        return Path(flow_cell_directory, DemultiplexingDirsAndFiles.DEMUX_COMPLETE).exists()
 
+    def update_sample_read_counts(self, sample_internal_ids: List[str]) -> None:
+        """Update samples in status db with the sum of all read counts for the sample in the sequencing metrics table."""
         for sample_id in sample_internal_ids:
-            LOG.info(f"Updating read count for sample {sample_id}")
-            sample: Optional[Sample] = self.status_db.get_sample_by_internal_id(
-                internal_id=sample_id
-            )
+            self.update_single_sample_read_count(sample_id)
+        self.status_db.session.commit()
 
-            if not sample:
-                LOG.info(f"Sample {sample_id} not found in status db")
-                continue
+    def update_single_sample_read_count(self, sample_id: str) -> None:
+        sample = self.status_db.get_sample_by_internal_id(internal_id=sample_id)
 
-            sample_read_count: int = self.status_db.get_number_of_reads_for_sample_from_metrics(
+        if sample:
+            sample_read_count = self.status_db.get_number_of_reads_for_sample_from_metrics(
                 sample_internal_id=sample_id
             )
-
             LOG.info(f"Updating sample {sample_id} with read count {sample_read_count}")
-
             sample.reads = sample_read_count
 
-        self.status_db.session.commit()
-
     def add_flow_cell_data_to_housekeeper(
         self, flow_cell_name: str, flow_cell_directory: Path
     ) -> None:
```

Comment:
Why did you delete the doctring?

Reply:
I think it is due to that in this case the docstring is a repetition of the function name and violates DRY in this way as it adds no other knowledge than what can be understood from the function name.

I mention this in the coding guidelines in Atlas. That being said, we have not yet discussed on how to implement these things. So I am not sure what would be right to do at this point, i.e. keeping or deleting such docstrings